vr-wordle/src/App.test.js

```diff
@@ -1,20 +1,70 @@
 import { render, screen } from "@testing-library/react";
 
-import { useState, useRef } from "react";
-
-import ReactThreeTestRenderer from "@react-three/test-renderer";
-
-test("generate letters generates 26 boxes", async () => {
-  const alphabet = "abcdefghijklmnopqrstuvwxyz".split("");
-  // const letters = useRef(<group />);
-  // const [state, setState] = useState({
-  //   guesses: ["hello", "world", "vrdle", "wrdle", "crane", "cramp"],
-  //   answer: "cramp",
-  //   reset: false,
-  // });
-  // const renderer = await ReactThreeTestRenderer.create(
-  //   generateLetters(state, alphabet, letters)
-  // );
-  // const meshChildren = renderer.scene.children[0].allChildren;
-  // expect(meshChildren.length).toBe(26);
+import React, { useState, useRef } from "react";
+
+// import ReactThreeTestRenderer from "@react-three/test-renderer";
+
+import Grid, { Guess } from "./Components/Grid.js";
+
+test("Guess correctly outputs a five letter word with five blocks", () => {
+  const expected = [
+    {
+      side: 1,
+      id: "r",
+      submitted: true,
+      index: 0,
+      position: [-2.2, 0, 0],
+      state: "present",
+    },
+    {
+      side: 1,
+      id: "u",
+      submitted: true,
+      index: 1,
+      position: [-1.1, 0, 0],
+      state: "absent",
+    },
+    {
+      side: 1,
+      id: "p",
+      submitted: true,
+      index: 2,
+      position: [0, 0, 0],
+      state: "absent",
+    },
+    {
+      side: 1,
+      id: "e",
+      submitted: true,
+      index: 3,
+      position: [1.1, 0, 0],
+      state: "present",
+    },
+    {
+      side: 1,
+      id: "e",
+      submitted: true,
+      index: 4,
+      position: [2.2, 0, 0],
+      state: "correct",
+    },
+  ];
+
+  const test = Guess("rupee", "irate");
+  for (let child in test.props.children) {
+    expect(JSON.stringify(test.props.children[child].props)).toStrictEqual(
+      JSON.stringify(expected[child])
+    );
+  }
```

Comment:
Really smart, great testing! But why did you need to use JSON.stringify? Was there a bug?